velox/docs/functions/spark/decimal.rst

```diff
@@ -0,0 +1,13 @@
+=================
+Decimal functions
+=================
+
+Spark decimal functions can be used in decimal value computation.
+
+Value functions
+---------------
+
+.. spark:function:: unscaled_value(x) -> bigint
+
+    Return the unscaled bigint value of a short decimal ``x``, assuming it fits in a bigint.
```

Comment:
Is it possible that it doesn't fit? I assume no.

Reply:
In spark, only short decimal will call this function, so it will fit a bigint, you can check https://github.com/apache/spark/blob/2c9b25a90b57ca2095881f9de4f13bf820f9dac9/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/decimalExpressions.scala#L41

Reply:
```
/**
 * Return the unscaled Long value of a Decimal, assuming it fits in a Long.
 * Note: this expression is internal and created only by the optimizer,
 * we don't need to do type check for it.
 */
```

I'm wondering about such internal expressions / functions. Should these be part of Gluten rather than Velox?

Reply:
cc @rui-mo

Reply:
These functions, although used internally by Spark, can appear in expression calculations and need to be executed in the native engine. Therefore, they cannot be placed within Gluten.